soc/arm/nxp_kinetis/Kconfig.orig

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2016 Intel Corporation
+# Copyright (c) 2016, Freescale Semiconductor, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_KINETIS
+	bool
+	select HAS_SEGGER_RTT
+
+if SOC_FAMILY_KINETIS
+
+config SOC_FAMILY
+	string
+	default "nxp_kinetis"
+
+source "soc/arm/nxp_kinetis/*/Kconfig.soc"
+
+config SOC_PART_NUMBER
+	default SOC_PART_NUMBER_KINETIS_K2X if SOC_SERIES_KINETIS_K2X
+	default SOC_PART_NUMBER_KINETIS_K6X if SOC_SERIES_KINETIS_K6X
+	default SOC_PART_NUMBER_KINETIS_K8X if SOC_SERIES_KINETIS_K8X
+	default SOC_PART_NUMBER_KINETIS_KWX if SOC_SERIES_KINETIS_KWX
+	default SOC_PART_NUMBER_KINETIS_KL2X if SOC_SERIES_KINETIS_KL2X
+	default SOC_PART_NUMBER_KINETIS_KE1XF if SOC_SERIES_KINETIS_KE1XF
+	default SOC_PART_NUMBER_KINETIS_KV5X if SOC_SERIES_KINETIS_KV5X
+
+config HAS_OSC
+	bool
+	help
+	  Set if the oscillator (OSC) module is present in the SoC.
+
+config HAS_MCG
+	bool
+	help
+	  Set if the multipurpose clock generator (MCG) module is present in the SoC.
+
+if HAS_OSC
+
+choice
+	prompt "Oscillator Mode Selection"
+	default OSC_EXTERNAL
+
+config OSC_EXTERNAL
+	bool "External reference clock"
+	help
+	  Set this option to use the oscillator in external reference clock mode.
+
+config OSC_LOW_POWER
+	bool "Low power oscillator"
+	help
+	  Set this option to use the oscillator in low-power mode.
+
+config OSC_HIGH_GAIN
+	bool "High gain oscillator"
+	help
+	  Set this option to use the oscillator in high-gain mode.
+
+endchoice
+
+config OSC_XTAL0_FREQ
+	int "External oscillator frequency"
```

Comment:
Does it make sense to change this in the `menuconfig`? If not, it's better to make it invisible (remove the prompt) and set it in a `Kconfig.defconfig` file.

Reply:
Changing this will affect other nxp kinetis based soc. Current implementation is aligned with other SOCs. IMHO I would mantain it as it is.